src/Casdoor.Client/CasdoorUserClient.cs

```diff
@@ -1,37 +1,148 @@
+        string str = await _httpClient.DoGetStringAsync(url);
+
+        using MemoryStream ms = new(Encoding.UTF8.GetBytes(str));
+        CasdoorUser[] users = await JsonSerializer.DeserializeAsync<CasdoorUser[]>(ms)
+                              ?? throw new CasdoorException("unable to deserialize json");
+        return users;
     }
 
-    public virtual Task<IEnumerable<CasdoorUser>> GetUsersAsync() => throw new NotImplementedException();
+    public virtual async Task<CasdoorUser> GetUserAsync(string name)
+    {
+        Dictionary<string, string> queryMap = new() {{"id", string.Concat(_options.OrganizationName, "/", name)}};
+
+        string url = _httpClient.GetUrl("get-user", queryMap);
+        string str = await _httpClient.DoGetStringAsync(url);
+
+        using MemoryStream ms = new(Encoding.UTF8.GetBytes(str));
+        CasdoorUser users = await JsonSerializer.DeserializeAsync<CasdoorUser>(ms)
+                            ?? throw new CasdoorException("unable to deserialize json");
+        return users;
+    }
 
-    public virtual Task<IEnumerable<CasdoorUser>> GetSortedUsersAsync(string sorter, int limit) => throw new NotImplementedException();
+    public virtual async Task<CasdoorUser> GetUserByEmailAsync(string email)
+    {
+        Dictionary<string, string> queryMap = new() {{"owner", _options.OrganizationName}, {"email", email},};
 
-    public virtual Task<CasdoorUser> GetUserAsync(string name) => throw new NotImplementedException();
+        string url = _httpClient.GetUrl("get-user", queryMap);
+        string str = await _httpClient.DoGetStringAsync(url);
 
-    public virtual Task<CasdoorUser> GetUserByEmailAsync(string email) => throw new NotImplementedException();
+        using MemoryStream ms = new(Encoding.UTF8.GetBytes(str));
+        CasdoorUser users = await JsonSerializer.DeserializeAsync<CasdoorUser>(ms)
+                            ?? throw new CasdoorException("unable to deserialize json");
+        return users;
+    }
 
-    public virtual Task<bool> AddUserAsync(CasdoorUser user) => throw new NotImplementedException();
+    public virtual async Task<bool> AddUserAsync(CasdoorUser user)
+    {
+        (_, bool ok) = await _httpClient.ModifyUserAsync("add-user", user, null);
+        return ok;
+    }
 
-    public virtual Task<bool> UpdateUserAsync(CasdoorUser user, params string[] propertyNames) => throw new NotImplementedException();
+    public virtual async Task<bool> UpdateUserAsync(CasdoorUser user, params string[] propertyNames)
+    {
+        (_, bool ok) = await _httpClient.ModifyUserAsync("update-user", user, propertyNames);
+        return ok;
+    }
 
-    public virtual Task<bool> DeleteUserAsync(string name) => throw new NotImplementedException();
+    public virtual async Task<bool> DeleteUserAsync(string name)
+    {
+        var user = await GetUserAsync(name);
+        (_, bool ok) = await _httpClient.ModifyUserAsync("delete-user", user, null);
```

Comment:
Do not use the value tuple here, we can throw an exception in ModifyUserAsync and provide `TryXXXAsync` to return the bool result later.. because we need to ensure that the API behaves consistently.
Creating a new value tuple instance has an unnecessary allocation.

Reply:
TryModifyUserAsync is not necessary? After all, `ok` is just a signal of `resp.Data is "Affected"`, which is borrowed from the go sdk.

**Possible solutions**
1. Check the `resp.Data` in `ModifyUserAsync` and throw a CasdoorApiException
   ```csharp
   if (resp.Data is not "Affected")
   {
       throw new CasdoorApiException("unsuccessful modification");
   }
   ```
2. Straightly pass the resp (CasdoorResponse) and let the caller do this check. In general, throwing exceptions comes with extra cost.

Reply:
Ok, I think we can provide two sets of APIs for different users to choose one form.
```csharp
public Task<CasdoorResponse> DeleteUserAsync(string name);
public Task<bool> TryDeleteUserAsync(string name);
```